setup.py

```diff
@@ -0,0 +1,176 @@
+"""The setup module for pysimulationio
+"""
+
+from setuptools import setup, find_packages
+from setuptools.extension import Extension
+from distutils.sysconfig import get_python_lib, get_python_inc
+from distutils.command.build import build
+from setuptools.command.install import install
+from distutils.command.clean import clean
+from os import path, environ
+import os, glob
+from shutil import rmtree
+
+# access some simple stuff
+here = path.abspath(path.dirname(__file__))
+py_include_dirs=list(set([get_python_inc(i) for i in range(2)]
+                         +[get_python_lib(i,j)\
+                           for i,j in zip(range(2),range(2))]))
+
+# Read in the package information generated by the makefile.
+with open('VERSION','r') as f:
+    version = float(f.read().strip())
+
+with open('README.md','r') as f:
+    long_description = f.read()
+
+try:
+    with open('includes.txt','r') as f:
+        includes = [x.strip() for x in f.readlines()]
+    includes += py_include_dirs
+    includes = [path.realpath(x) for x in includes]
+
+    with open('flags.txt','r') as f:
+        flags = f.read().strip().split(' ')
+
+    with open('links.txt','r') as f:
+        links = [x.strip() for x in f.readlines()]
+    links = [path.realpath(x) for x in links]
+    link_args = ["-Wl,-rpath,{}".format(p) for p in links]
+
+    with open('compiler.txt','r') as f:
+        cc = f.read().strip()
+    environ["CC"] = cc
+
+    with open('libs.txt','r') as f:
+        libs = [x. strip() for x in f.readlines()]
+    libs = reduce(lambda x,y: x+y,[x.split(' ') for x in libs])
+    libs = [x.lstrip('-l') for x in libs]
+
+    with open('lib_sources.txt','r') as f:
+        lib_sources = f.read().strip().split(' ')
+except IOError:
+    raise IOError("There is missing metadata. Try running "
+                  +"'make meta' to generate it.")
+
+
+all_object_files = glob.glob(path.join(here,'*.o'))
+if len(all_object_files) == 0:
+    raise IOError("There are missing object files! Try running "
+                  +"'make' to build SimulationIO.")
+object_files = [src.replace('cpp','o') for src in lib_sources]
```

Comment:
If you have a source file called `acppi.cpp`, then you will calculate an object file name of `aoi.o`. That won't work. You need to make sure only the suffix is replaced, not any occurrence of `cpp`. Hence: Look for `.cpp` instead of just `cpp`.